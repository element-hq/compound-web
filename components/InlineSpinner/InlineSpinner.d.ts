import { default as React } from '../../../node_modules/react';
export declare const InlineSpinner: React.ForwardRefExoticComponent<{
    size?: number;
} & React.SVGAttributes<SVGElement> & React.RefAttributes<SVGSVGElement>>;
//# sourceMappingURL=InlineSpinner.d.ts.map