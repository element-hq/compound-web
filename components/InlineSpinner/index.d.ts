export { InlineSpinner } from './InlineSpinner';
//# sourceMappingURL=index.d.ts.map