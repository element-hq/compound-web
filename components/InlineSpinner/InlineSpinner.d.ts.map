{"version":3,"file":"InlineSpinner.d.ts","sourceRoot":"","sources":["../../../src/components/InlineSpinner/InlineSpinner.tsx"],"names":[],"mappings":"AAOA,OAAO,KAAoC,MAAM,OAAO,CAAC;AAQzD,eAAO,MAAM,aAAa;WAHjB,MAAM;yEAcd,CAAC"}