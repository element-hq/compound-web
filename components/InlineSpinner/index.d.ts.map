{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/InlineSpinner/index.ts"],"names":[],"mappings":"AAOA,OAAO,EAAE,aAAa,EAAE,MAAM,iBAAiB,CAAC"}