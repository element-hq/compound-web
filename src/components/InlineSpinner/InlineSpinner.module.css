/*
Copyright 2024 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

@keyframes spin {
  from {
    transform: rotateZ(0deg);
  }

  to {
    transform: rotateZ(360deg);
  }
}

.icon {
  color: var(--cpd-color-icon-secondary);
  display: inline-flex;
  justify-content: center;
  align-items: center;
  inline-size: 100%;
  block-size: 100%;
  animation: 1s linear spin infinite;
}
