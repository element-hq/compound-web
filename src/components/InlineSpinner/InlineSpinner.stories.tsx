/*
Copyright 2024 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

import { Meta, StoryObj } from "@storybook/react-vite";

import { InlineSpinner as InlineSpinnerComponent } from "./InlineSpinner";

const meta = {
  title: "InlineSpinner",
  component: InlineSpinnerComponent,
  tags: ["autodocs"],
  argTypes: {
    size: {
      type: "number",
    },
  },
  args: {},
} satisfies Meta<typeof InlineSpinnerComponent>;
export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
  parameters: {},
};
