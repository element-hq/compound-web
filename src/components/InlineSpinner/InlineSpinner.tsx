/*
Copyright 2024 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

import React, { SVGAttributes, forwardRef } from "react";
import styles from "./InlineSpinner.module.css";
import SpinnerIcon from "@vector-im/compound-design-tokens/assets/web/icons/spinner";

type InlineSpinnerProps = {
  size?: number;
} & SVGAttributes<SVGElement>;

export const InlineSpinner = forwardRef<SVGSVGElement, InlineSpinnerProps>(
  function InlineSpinner({ size = 20, ...props }, ref) {
    return (
      <SpinnerIcon
        ref={ref}
        className={styles.icon}
        style={{ width: size, height: size }}
        {...props}
      />
    );
  },
);
