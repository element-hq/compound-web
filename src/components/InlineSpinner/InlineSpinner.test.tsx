/*
Copyright 2024 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import React from "react";

import { InlineSpinner } from "./InlineSpinner";

describe("InlineSpinner", () => {
  it("renders", () => {
    const { asFragment } = render(<InlineSpinner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
