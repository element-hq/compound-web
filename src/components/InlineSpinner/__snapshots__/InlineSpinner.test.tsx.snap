// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InlineSpinner > renders 1`] = `
<DocumentFragment>
  <svg
    class="_icon_01cd2d"
    fill="currentColor"
    height="1em"
    style="width: 20px; height: 20px;"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M12 4.031a8 8 0 1 0 8 8 1 1 0 0 1 2 0c0 5.523-4.477 10-10 10s-10-4.477-10-10 4.477-10 10-10a1 1 0 1 1 0 2"
      fill-rule="evenodd"
    />
  </svg>
</DocumentFragment>
`;
